packages/block-library/src/image/edit.js

```diff
@@ -321,7 +291,7 @@ export function ImageEdit( {
 	const shadowProps = getShadowClassesAndStyles( attributes );
 
 	const classes = classnames( className, {
-		'is-transient': temporaryURL,
+		'is-transient': temporaryURL || isTemp,
```

Comment:
Why do we need the `isTemp`? How is different from `temporaryURL`?

Reply:
The “onmount” upload won’t set blob URL as temp state as URL attribute is already a blob.

Decided to keep temp URL state for other upload methods.

Reply:
Sorry, but still don't get it. Can you elaborate a bit? 
Is the problem for setting the initial state when mounting in case of drag and drop in the editor?
Could this be resolved by this:
```
const [ temporaryURL, setTemporaryURL ] = useState(
	isTemporaryImage( id, url )
);
```

Reply:
I'll try my best.

* The image block is mounted by transformation (the file type).
* The `url` attribute is Blob, so we already have a temporary URL; there's no need to rerender the component once more to store its copy in the state.
* When the on-mount upload is complete, the block will receive actual media data.

We can set the initial `temporaryURL` state on the mount, but I don't think it's needed at this point.

Reply:
Update logic in [bd8cad8](https://github.com/WordPress/gutenberg/pull/60208/commits/bd8cad8ed05d15ee8a4966f87c700e6edb7b3696).